### PR TITLE
extract out binary and command module

### PR DESCRIPTION
**Pull request by dannypsnl**
NOTE: @YanHerChen notices the below reference text

> Cargo will also treat any files located in src/bin/*.rs as executables.

### CODE REVIEW

File: src/bin/cmd/compile.rs [hunk 1]
```diff
@@ -1,8 +1,8 @@
-use crate::ast::TopAst;
-use crate::codegen::llvm::LLVMValue;
-use crate::codegen::CodeGenerator;
-use crate::parser::{parse_prelude, Parser};
-use crate::semantic::SemanticChecker;
+use elz::ast::TopAst;
+use elz::codegen::llvm::LLVMValue;
+use elz::codegen::CodeGenerator;
+use elz::parser::{parse_prelude, Parser};
+use elz::semantic::SemanticChecker;
```
Comment (YanHerChen) [inline src/bin/cmd/compile.rs:5]: what is difference between `use crate` and `use elz`?

**Comment by dannypsnl** [comment]
the result is not good, close now

File: src/bin/elz.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 use clap::{App, Arg, SubCommand};
-use elz::cmd;
+
+mod cmd;
```
Comment (YanHerChen) [inline src/bin/elz.rs:3]: what's purpose of `mod cmd`?

File: src/codegen/formatted_elz/mod.rs [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+pub const DEFAULT_LEVEL: usize = 0;
 const INDENT: &'static str = "  ";
 
-pub(crate) trait FormattedElz {
+pub trait FormattedElz {
     fn formatted_elz(&self, level: usize) -> String;
 }
 
-pub(crate) struct FormatTopAstList(pub(crate) Vec<TopAst>);
+pub struct FormatTopAstList(pub Vec<TopAst>);
```
Comment (YanHerChen) [inline src/codegen/formatted_elz/mod.rs:10]: why delete `(crate)`?

**Pull request closed without merging**